### PR TITLE
Fix for order not confirmed issue

### PR DESCRIPTION
**Pull request by enishant**
* Update checkout timeout
* Process order which has order status as cancel and state as new

### CODE REVIEW

**Review by nikhilsharma1100 — APPROVED** [review]

File: Controller/Payment/Webhook.php [hunk 1]
```diff
@@ -237,8 +238,11 @@ protected function authorize(array $post)
                 {
                     $payment = $order->getPayment();
 
-                    if (($order->getStatus() === static::STATUS_PENDING) and
-                        (empty($payment->getLastTransId()) === true) )
+                    if ((($order->getState() === static::STATE_NEW
+                        && $order->getStatus() === static::STATUS_CANCELED)
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant please use `and` and `or`

File: Controller/Payment/Webhook.php [hunk 2]
```diff
@@ -333,13 +337,16 @@ protected function orderPaid(array $post)
...
-                    if (in_array($order->getStatus(), [static::STATUS_PENDING, static::STATUS_PROCESSING]))
+                    if (in_array($order->getStatus(), [static::STATUS_PENDING, static::STATUS_PROCESSING, static::STATUS_CANCELED]))
                     {
                         $payment = $order->getPayment();
 
                         $amountPaid = number_format($rzpOrderAmount / 100, 2, ".", "");
 
-                        if ($order->getStatus() === static::STATUS_PENDING)
+                        if ($order->getStatus() === static::STATUS_PENDING
+                            || ($order->getState() === static::STATE_NEW
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant,  formatting issue

Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05  updated condition and fixed formatting issue.

File: Controller/Payment/Webhook.php (hunk 1 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05  updated condition and fixed formatting issue.

File: Controller/Payment/Webhook.php [hunk 3]
```diff
@@ -472,4 +537,22 @@ protected function getPostData() : array
         
         return json_decode($request, true);
     }
+
+    protected function getOrderWebhookData($orderId) : array
+    {
+        $collection = $this->objectManagement->get('Magento\Sales\Model\Order')
+        ->getCollection()
```
Comment (nikhilsharma1100) [inline Controller/Payment/Webhook.php]: Please follow the chaining functions, alignment formatting rules

File: Setup/UpgradeSchema.php [hunk 4]
```diff
@@ -41,20 +41,27 @@ public function upgrade(SchemaSetupInterface $setup,
...
-                    [
-                        'nullable' => true,
-                        'type'     => \Magento\Framework\DB\Ddl\Table::TYPE_TEXT,
-                        'length'   => 55,
-                        'comment'  => 'RZP Order ID'
-                    ]
-                );
-            }
+        $tableName = $setup->getTable('sales_order');
+        if ($setup->getConnection()->isTableExists($tableName) == true) {
```
Comment (nikhilsharma1100) [inline Setup/UpgradeSchema.php]: Keep the opening braces for this if statement in new line

File: Controller/Payment/Webhook.php [hunk 5]
```diff
@@ -163,6 +174,31 @@ public function execute()
...
+                    {
+                        $this->setWebhookNotifiedAt($orderWebhookData['entity_id']);
+
+                        $this->logger->info("Razorpay Webhook: Updated WebhookNotifiedAt.");
+
+                        header('Status: ' . static::HTTP_CONFLICT_STATUS . ' Webhook conflicts due to early execution.', true, static::HTTP_CONFLICT_STATUS);
+                        exit;
+                    }
+                    elseif(!empty($orderWebhookData['rzp_webhook_notified_at']) === true and
+                        ((time() - $orderWebhookData['rzp_webhook_notified_at']) < static::WEBHOOK_NOTIFY_WAIT_TIME)
```
Comment (nikhilsharma1100) [inline Controller/Payment/Webhook.php]: This conditional block needs to be aligned with above condition

File: Controller/Payment/Webhook.php [hunk 6]
```diff
@@ -237,9 +273,17 @@ protected function authorize(array $post)
...
-                    if (($order->getStatus() === static::STATUS_PENDING) and
-                        (empty($payment->getLastTransId()) === true) )
+                    if ($order->getState() === static::STATE_NEW and
+                        ($order->getStatus() === static::STATUS_CANCELED or
+                        $order->getStatus() === static::STATUS_PENDING)
+                    )
                     {
+
+                        $this->logger->info("Razorpay Webhook: "
+                        . " Event: " . $post['event']
```
Comment (nikhilsharma1100) [inline Controller/Payment/Webhook.php]: Please follow the multiple arguments, alignment formatting rules

File: Controller/Payment/Webhook.php [hunk 7]
```diff
@@ -333,13 +382,24 @@ protected function orderPaid(array $post)
...
                 if ($order)
                 {
-                    if (in_array($order->getStatus(), [static::STATUS_PENDING, static::STATUS_PROCESSING]))
+                    if (in_array($order->getStatus(), [static::STATUS_PENDING, static::STATUS_PROCESSING]) or
+                        ($order->getState() === static::STATE_NEW and
+                         $order->getStatus() === static::STATUS_CANCELED)
+                    )
                     {
+                        $this->logger->info("Razorpay Webhook: "
+                        . " Event: " . $post['event']
```
Comment (nikhilsharma1100) [inline Controller/Payment/Webhook.php]: Please follow the multiple arguments, alignment formatting rules

File: Controller/Payment/Webhook.php (hunk 3 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @nikhilsharma1100 Updated formatting for chaining functions

File: Setup/UpgradeSchema.php (hunk 4 above)
Comment (enishant) [reply Setup/UpgradeSchema.php]: @nikhilsharma1100 Updated opening braces for if statement in new line

File: Controller/Payment/Webhook.php (hunk 5 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @nikhilsharma1100 Conditional block aligned with above condition

File: Controller/Payment/Webhook.php (hunk 6 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @nikhilsharma1100 Updated alignment formatting for the multiple arguments

File: Controller/Payment/Webhook.php (hunk 7 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @nikhilsharma1100 Updated alignment formatting for the multiple arguments

**Review by nikhilsharma1100 — APPROVED** [review]

**Comment by nikhilsharma1100** [comment]
LGTM

File: Controller/Payment/Webhook.php [hunk 8]
```diff
@@ -237,9 +273,17 @@ protected function authorize(array $post)
                 {
                     $payment = $order->getPayment();
 
-                    if (($order->getStatus() === static::STATUS_PENDING) and
-                        (empty($payment->getLastTransId()) === true) )
+                    if ($order->getState() === static::STATE_NEW and
```
Comment (ramth05) [inline Controller/Payment/Webhook.php:276]: @enishant , still formatting not correct 
if (($order->getState() === static::STATE_NEW) and
    (($order->getStatus() === static::STATUS_CANCELED) or
     ($order->getStatus() === static::STATUS_PENDING)))

Comment (ramth05) [reply Controller/Payment/Webhook.php:276]: Please apply where it's not correct.

File: Controller/Payment/Webhook.php [hunk 9]
```diff
@@ -163,6 +174,31 @@ public function execute()
...
+                    if (!empty($orderWebhookData['rzp_webhook_notified_at']) === false)
+                    {
+                        $this->setWebhookNotifiedAt($orderWebhookData['entity_id']);
+
+                        $this->logger->info("Razorpay Webhook: Updated WebhookNotifiedAt.");
+
+                        header('Status: ' . static::HTTP_CONFLICT_STATUS . ' Webhook conflicts due to early execution.', true, static::HTTP_CONFLICT_STATUS);
+                        exit;
+                    }
+                    elseif (!empty($orderWebhookData['rzp_webhook_notified_at']) === true and
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , why `!`

File: Controller/Payment/Webhook.php [hunk 10]
```diff
@@ -163,6 +174,31 @@ public function execute()
                     header('Status: 400 Signature Verification failed', true, 400);
                     exit;
                 }
+
+                if (isset($post['payload']['payment']['entity']['notes']['merchant_order_id']) === true)
+                {
+                    $orderId = $post['payload']['payment']['entity']['notes']['merchant_order_id'];
+                    $orderWebhookData = $this->getOrderWebhookData($orderId);
+                    if (!empty($orderWebhookData['rzp_webhook_notified_at']) === false)
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , why `!`

File: Controller/Payment/Webhook.php (hunk 9 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Updated condition check

File: Controller/Payment/Webhook.php (hunk 10 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Updated condition check

**Pull request merged**